### PR TITLE
Ciao Jenkins!

### PR DESCRIPTION
**Pull request by amCap1712**
We wish to reduce the maintenance due to our self hosted jenkins instance. Our experience with Travis CI has been rather dismal. Hence, trying Github Actions.
 
Related: https://github.com/metabrainz/brainzutils-python/pull/63

### CODE REVIEW

File: .github/workflows/spark-tests.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      continue-on-error: true
+
+    - name: Pull docker images
+      run: docker-compose -f docker/docker-compose.spark.test.yml pull
+
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+
+    - name: Build the Docker image
+      run: docker-compose -f docker/docker-compose.spark.test.yml -p listenbrainz_spark_test build
```
Comment (alastair) [inline .github/workflows/spark-tests.yml:32]: how much of these commands replicate what we have in test.sh scripts? Would be neat to try and reuse them

**Review by alastair — COMMENTED** [review]
Some questions that occurred to me to answer before we go ahead with the merge

File: .github/workflows/frontend-tests.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Login to Docker Hub
+      run: echo ${{ secrets.DOCKER_HUB_PASSWORD }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/frontend-tests.yml:19]: Yes.

File: .github/workflows/frontend-tests.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: ListenBrainz Frontend Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/frontend-tests.yml:7]: No. We can change the branches here to '*' to run against PRs against all branches.

File: .github/workflows/frontend-tests.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      run: docker-compose -f docker/docker-compose.test.yml pull
+
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+
+    - name: Build the Docker image
+      run: docker-compose -f docker/docker-compose.test.yml -p listenbrainz_frontend_test build frontend_tester
+
+    - name: Run frontend tests
+      run: docker-compose -f docker/docker-compose.test.yml -p listenbrainz_frontend_test run frontend_tester npm run test:ci || true
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/frontend-tests.yml:32]: These are still reported as failures when you click the test but yeah the summary would it as passing.

[the remark this answers is not in the document]
Comment (alastair) [reply .github/workflows/frontend-tests.yml:32]: This is where reusing a test.sh script could help - it'll let us run all 3 commands, but then return a failing state from the script if any one of the commands fails

File: .github/workflows/integration-tests.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Create configuration file
+      run: cp listenbrainz/config.py.sample listenbrainz/config.py
+
+    - name: Login to Docker Hub
+      run: echo ${{ secrets.DOCKER_HUB_PASSWORD }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
+      continue-on-error: true
+
+    # by default, docker-compose tries to pull the `listenbrainz` image used by timescale_writer as well.
+    # but this is built locally for the listenbrainz service and reused for timescale writer, hence the pull fails.
+    # to avoid this we specifically mention the services to pull images for here,
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/integration-tests.yml:27]: This is locally reproducible by running `docker-compose -f docker/docker-compose.integration.yml pull` in the repository root.

File: .github/workflows/frontend-tests.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/frontend-tests.yml]: Done.

File: .github/workflows/frontend-tests.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (amCap1712) [reply .github/workflows/frontend-tests.yml]: Yes, that's possible. Alternatively, I recently discovered there's a `if: ${{ always() }}` clause available for this use case and seems simpler.

File: .github/workflows/spark-tests.yml (hunk 1 above)
Comment (amCap1712) [reply .github/workflows/spark-tests.yml]: Fixed.

**Review by alastair — APPROVED** [review]
🥇

**Pull request merged**